source/common/tls/server_context_impl.cc

```diff
@@ -42,18 +42,24 @@
 namespace Envoy {
 namespace {
 
-bool cbsContainsU16(CBS& cbs, uint16_t n) {
+Ssl::CurveNIDSupportedVector getClientCurveNIDSupported(CBS& cbs) {
+  Ssl::CurveNIDSupportedVector cnsv{};
   while (CBS_len(&cbs) > 0) {
     uint16_t v;
     if (!CBS_get_u16(&cbs, &v)) {
-      return false;
+      return cnsv;
```

Comment:
nit: slightly simpler control flow if there is only 1 return in the function
```suggestion
      break;
```